mlyacc/src/utils.sml

```diff
@@ -512,20 +512,20 @@ signature HASH =
 (* hash: creates a hash table of size n which assigns each distinct member
    a unique integer between 0 and n-1 *)
 
-functor Hash(B : sig type elem
+functor HashTable(B : sig type elem
```

Comment:
Again, we shouldn't need to rename this structure because of an addition in the MLton Library.  You should be able to fix the naming clash by only changing `/mlyacc/mlyacc.mlb`.  (Again, probably the right thing to do is add an export filter to the use of the MLton Library to only expose the modules that are used by `main.sml`.)  In fact, I would suggest rebasing (and force pushing) the PR so as not to have any commits touching these files.